e2e/cypress/integration/lms_video_spec.js

```diff
@@ -0,0 +1,157 @@
+
+  before(() => {
+    cy.lmsLoginAdmin();
+    // Navigate to the courseware.
+    // Triggers `video_player_ready` event.
+    cy.visit(sectionUrl);
+
+    // Wait for the video to appear and set video src and type.
+    // Triggers `load_video` event.
+    cy.get("video")
+      .first()
+      .should("exist")
+      .invoke("attr", "src", videoAsset.url)
+      .should("have.attr", "src", videoAsset.url)
+      .invoke("attr", "type", videoAsset.content_type)
+      .should("have.attr", "type", videoAsset.content_type);
+
+    // Click video play button.
+    // Triggers `play_video` event.
+    cy.get("video")
+      .first()
+      .should("have.prop", "paused", true)
+      .and("have.prop", "ended", false)
+      .then(($video) => {
+        $video[0].play();
+      });
+
+    // Click video speed button.
+    // Triggers `speed_change_video` event.
+    cy.get("video")
+      .first()
+      .should("have.prop", "paused", false)
+      .then(($video) => {
+        $video[0].playbackRate = 2;
+      });
+
+    // Seek to the first second of the video.
+    // Triggers `seek_video` event.
+    cy.get("video")
+      .first()
+      .and("have.prop", "playbackRate", 2)
+      .then(($video) => {
+        $video[0].currentTime = 1;
+      });
+
+    // Watch the whole 5 seconds video (in x2 speed).
+    // Triggers `pause_video` and `stop_video` events.
+    cy.get("video").first().should("have.prop", "ended", true);
+
+    // Other video interactions require the video to be loaded with `window.Video()`.
+    const metadata = {
+      autohideHtml5: "true",
+      autoplay: "false",
+      captionDataDir: "",
+      endTime: "",
+      generalSpeed: "1.0",
+      saveStateUrl: "",
+      savedVideoPosition: "0",
+      showCaptions: "false",
+      sources: ["/asset-v1:org+number+session+type@asset+block@video.mp4"],
+      speed: "1.0",
+      startTime: "",
+      streams: "",
+      sub: "",
+      transcriptAvailableTranslationsUrl: "",
+      transcriptLanguage: "en",
+      transcriptLanguages: { en: "English" },
+      transcriptTranslationUrl: "",
+      ytApiUrl: "",
+      ytImageUrl: "",
+      ytTestTimeout: "1500",
+      ytMetadataUrl: "",
+    };
+    cy.get("video")
+      .first()
+      .invoke("attr", "data-metadata", JSON.stringify(metadata))
+      .invoke("addClass", "video");
+
+    cy.window().then((win) => {
+      win.Video(".videoplayer");
+      win
+        .$("video")
+        .first()
+        // Triggers `show_transcript` event.
+        .trigger("captions:show")
+        // Triggers `hide_transcript` event.
+        .trigger("captions:hide")
+        // Triggers `video_show_cc_menu` event.
+        .trigger("language_menu:show")
+        // Triggers `video_hide_cc_menu` event.
+        .trigger("language_menu:hide")
+        // Triggers `skip_video` event.
+        .trigger("skip")
+        // Triggers `do_not_show_again_video` event.
+        .trigger("skip", true);
```

Comment:
Is it necessary to trigger these two events ?

Reply:
The test are checking the presence of `skip_video` and `do_not_show_again_video` edX video events. Without these triggers, the two tests fail. In the FUN version, I haven't found a way to properly generate those events. Will try the `bare` edX version in the future to double-check.